### ISSUE TITLE
search_is_some incorrect text

### ISSUE THREAD

**Issue by xNxExOx** [rust-lang/rust-clippy#11681]
### Summary

Help suggestion is wrong.
It say to use `any` instead of `any`, but the code does not contains `any`.

### Reproducer

I tried this code:

```rust
 let _ = [1, 2, 3].iter().find(|x| **x == 3).is_some();
```

I expected to see this happen:
I did not really think about using `any`, so I did not expect an warning.

Instead, this happened:
```
warning: called `is_some()` after searching an `Iterator` with `find`
 --> src/main.rs:4:30
  |
4 |     let _ = [1, 2, 3].iter().find(|x| **x == 3).is_some();
  |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `any()` instead: `any(|x| *x == 3)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#search_is_some
  = note: `#[warn(clippy::search_is_some)]` on by default
```

### Version

```text
rustc 1.72.0 (5680fa18f 2023-08-23)
binary: rustc
commit-hash: 5680fa18feaa87f3ff04063800aec256c3d4b4be
commit-date: 2023-08-23
host: x86_64-pc-windows-msvc
release: 1.72.0
LLVM version: 16.0.5
```

**Comment by llogiq** [comment]
The suggestion is correct. You missed the colon, which means the following is *how* to use `any` instead.

**Comment by xNxExOx** [comment]
I do not know, if the sentence is technically correct in UK English, or US English, or AU English, or any other subset of English, but to me it is really confusing. Something like `` help: consider using: `any(|x| *x == 3)` instead `` would make much more sense.

Even just `` help: use `any()` `` would be less confusing that the current wording.

**Comment by GuillaumeGomez** [comment]
We use `consider using` in a few places, so I think we should here as well. Sending a fix.

**Comment by GuillaumeGomez** [comment]
I opened #12140.
